server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java

```diff
@@ -0,0 +1,269 @@
+import au.org.democracydevelopers.corla.util.testUtils;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.google.gson.Gson;
+import org.apache.http.HttpStatus;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.jdbc.JdbcDatabaseDelegate;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import spark.HaltException;
+import spark.Request;
+import us.freeandfair.corla.Main;
+import us.freeandfair.corla.asm.*;
+import us.freeandfair.corla.controller.ContestCounter;
+import us.freeandfair.corla.model.AuditReason;
+import us.freeandfair.corla.model.Choice;
+import us.freeandfair.corla.model.ContestResult;
+import us.freeandfair.corla.persistence.Persistence;
+
+import javax.transaction.Transactional;
+import java.util.*;
+
+import static au.org.democracydevelopers.corla.endpoint.AbstractAllIrvEndpoint.RAIRE_URL;
+import static au.org.democracydevelopers.corla.endpoint.GenerateAssertions.CONTEST_NAME;
+import static au.org.democracydevelopers.corla.util.testUtils.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.testng.Assert.*;
+import static us.freeandfair.corla.asm.ASMEvent.DoSDashboardEvent.*;
+import static us.freeandfair.corla.endpoint.Endpoint.AuthorizationType.STATE;
+
+/**
+ * Test the GetAssertions endpoint via the API.
+ * This currently tests that the assertion generation request is accepted and blocked in the right
+ * circumstances.
+ * TODO This really isn't a completely comprehensive set of tests yet.
+ * See <a href="https://github.com/DemocracyDevelopers/colorado-rla/issues/125">...</a>
+ */
+public class GenerateAssertionsAPITests extends TestClassWithAuth {
+
+  /**
+   * Class-wide logger.
+   */
+  private static final Logger LOGGER = LogManager.getLogger(GenerateAssertionsAPITests.class);
+
+  /**
+   * Container for the mock-up database.
+   */
+  private final static PostgreSQLContainer<?> postgres = createTestContainer();
+
+  /**
+   * The Generate Assertions endpoint.
+   */
+  private final GenerateAssertions endpoint = new GenerateAssertions();
+
+  /**
+   * Mock response for tinyExample1 contest
+   */
+  private final static GenerateAssertionsResponse tinyIRVResponse
+      = new GenerateAssertionsResponse(tinyIRV, true, false);
+
+  /**
+   * Request for tinyExample1 contest
+   */
+  private final static GenerateAssertionsRequest tinyIRVRequest
+      = new GenerateAssertionsRequest(tinyIRV, tinyIRVCount, 5,
+      tinyIRVCandidates.stream().map(Choice::name).toList());
+
+  /**
+   * Raire endpoint for getting assertions.
+   */
+  private final String raireGenerateAssertionsEndpoint = "/raire/generate-assertions";
+
```

Comment:
Can the port be specified somewhere else in settings ... what if 8110 clashes with something else someone is running?

Reply:
Excellent idea; done in the other PR.